frontend/src/metabase-lib/expressions/diagnostics.ts

```diff
@@ -151,8 +163,17 @@ function prattCompiler(source, startRule, legacyQuery, name) {
     }
   } catch (err) {
     console.warn("compile error", err);
-    return err;
+    return err as ErrorWithMessage;
```

Comment:
We can avoid these casts by introducing type guards + having a default error.

Ccomponents use this default error: 
```
{ message: t`Invalid expression` }
```

Reply:
IDK, from what I know, error in catch blocks is untyped because of many reasons, so you have to manually cast it

Reply:
> error in catch blocks is untyped because of many reasons

That's precisely why they should not be casted.

```suggestion
    if (isErrorWithMessage(err)) {
      return err;
    }

    return { message: t`Invalid expression` };
```

Reply:
added type guard, thanks